core/controllers/release_coordinator.py

```diff
@@ -58,6 +59,117 @@ def delete(self) -> None:
         self.render_json({})
 
 
+class UserGroupHandlerNormalizePayloadDict(TypedDict):
+    """Dict representation of UserGroupHandler's normalized_payload
+    dictionary.
+    """
+
+    action: str
+    user_group_name: Optional[str]
+    user_group_users: Optional[List[str]]
+    old_user_group_name: Optional[str]
+    user_group_to_delete: Optional[str]
+
+
+class UserGroupHandler(
+    base.BaseHandler[
+        UserGroupHandlerNormalizePayloadDict, Dict[str, str]]
+):
+    """Handler for user groups."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'action': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': [
+                        'update_user_group',
+                        'delete_user_group'
+                    ]
+                },
+                # TODO(#13331): Remove default_value when it is confirmed that,
+                # for clearing the search indices of exploration & collection
+                # 'action' field must be provided in the payload.
+                'default_value': None
+            },
+            'user_group_name': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'user_group_users': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'basestring'
+                    }
+                },
+                'default_value': None
+            },
+            'old_user_group_name': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'user_group_to_delete': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            }
+        }
+    }
+
+    @acl_decorators.can_access_release_coordinator_page
+    def get(self) -> None:
+        """Populates the data for user groups."""
+        user_group_models = user_services.get_all_user_groups()
+        all_users_usernames = user_services.get_all_users_usernames()
+
+        self.render_json({
+            'user_group_models': user_group_models,
+            'all_users_usernames': all_users_usernames
+        })
+
+    @acl_decorators.can_access_release_coordinator_page
+    def post(self) -> None:
+        """Performs series of action based on action parameter for user
+        groups.
+        """
+        assert self.user_id is not None
+        assert self.normalized_payload is not None
+        action = self.normalized_payload.get('action')
+        try:
+            if action == 'update_user_group':
```

Comment:
I think this isn't the right modelling. Use put() for updating and delete() for deleting, and pass simpler payloads -- you don't need an "action" field here, it's overkill.

Reply:
Updated!